### PR TITLE
feat: semantic viewer run query button

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/11031

from the ticket:

- [x] I can manually trigger a query:
- [x] the run query button has a row limit (max limit is 5,000 rows)
- [x] (no pagination for now, loads full table) the results table from the query is default scroll (not paginated)
- [x] (done in a separate PR) We need to think about streaming results here (similar to the SQL runner).
- [ ] (not going to be implemented here) The metrics look different to dimensions (e.g. blue/yellow if no one has a better idea).

bonus:
- [x] abstract semantic layer configurations and have an api call to get information about the semantic layer client
- [x] fetch semantic layer limit separately
- [x] introduces better state initialization

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/common/src/types/semanticLayer.ts [hunk 1]
```diff
@@ -112,14 +112,35 @@ export interface SemanticLayerTransformer<
...
 
+const SEMANTIC_LAYER_MAX_QUERY_LIMIT = 500;
+
 export function getDefaultedLimit(
     maxQueryLimit: number,
     queryLimit?: number,
 ): number {
-    return Math.min(queryLimit || 500, maxQueryLimit);
+    return Math.min(
+        queryLimit ?? SEMANTIC_LAYER_MAX_QUERY_LIMIT,
```
Comment (almeidabbm) [inline packages/common/src/types/semanticLayer.ts:122]:
```suggestion
        queryLimit ?? SEMANTIC_LAYER_DEFAULT_QUERY_LIMIT,
```

File: packages/frontend/src/features/semanticViewer/components/RunSemanticQueryButton.tsx [hunk 2]
```diff
@@ -100,20 +115,69 @@ export const RunSemanticQueryButton: FC = () => {
         sortBy,
     ]);
 
+    const handleSubmit = useCallback(
+        () =>
+            runSemanticViewerQuery({
+                ...allSelectedFieldsByKind,
+                sortBy,
```
Comment (almeidabbm) [inline packages/frontend/src/features/semanticViewer/components/RunSemanticQueryButton.tsx:122]: Missing limit here, when I change the limit I don't see it in the request payload

e.g. here I set the limit to 3 and I got all results

image


```suggestion
                sortBy,
                limit,
```

Comment (almeidabbm) [reply packages/frontend/src/features/semanticViewer/components/RunSemanticQueryButton.tsx:122]: Also on line 96 when we create the new transformer, should we be passing limit in the query?

**Review by almeidabbm — COMMENTED** [review]
Couple minor things, otherwise looks super!! :D

File: packages/common/src/types/semanticLayer.ts [hunk 3]
```diff
@@ -112,14 +112,35 @@ export interface SemanticLayerTransformer<
     sqlToString: (sql: SqlType) => string;
 }
 
+const SEMANTIC_LAYER_MAX_QUERY_LIMIT = 500;
```
[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/common/src/types/semanticLayer.ts:115]: I see, now it makes sense.

**Review by almeidabbm — APPROVED** [review]

**Pull request merged**